### PR TITLE
Chore: Remove `Stack`

### PR DESCRIPTION
**Pull request by thomasdigby**
Remove all trace of `Stack` and replace with `Flex`

### CODE REVIEW

File: lib/src/components/chip/ChipGroup.tsx [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 type TChipGroupProps = {
   gap: 1 | 2 | 3
 }
 
 export const ChipGroup: React.ForwardRefExoticComponent<
-  TChipGroupProps & React.ComponentProps<typeof Stack>
-> = React.forwardRef(({ gap = 2, ...rest }, ref) => {
-  return <Stack ref={ref} direction="row" gap={gap} align={false} {...rest} />
-})
+  TChipGroupProps & React.ComponentProps<typeof Flex> & { as?: any }
```
Comment (thomasdigby) [inline lib/src/components/chip/ChipGroup.tsx:10]: `as` is just not working for this component, I _think_ it's some side-effect of having double nested `forwardRef` components, but really 🤷

File: lib/src/components/notification-badge/NotificationBadge.tsx [hunk 2]
```diff
@@ -27,7 +27,7 @@ const StyledBadge = styled(Flex, {
   bg: '$primary800'
 })
 
-type NotificationBadgeProps = React.ComponentProps<typeof StyledBadge> & {
```
Comment (thomasdigby) [inline lib/src/components/notification-badge/NotificationBadge.tsx:30]: None of these props actually apply to the badge

File: documentation/content/components.progress-bar.md [hunk 3]
```diff
@@ -24,13 +24,13 @@ tabs:
       These are the available `themes` for this component: `Primary` (default), `Success`, `Warning`, `Danger`
 
 
-      <CodeBlock live={true} preview={true} code={`<Stack gap={3} css={{width: '100%'}}>
+      <CodeBlock live={true} preview={true} code={`<Flex gap={3} direction="column" css={{ width: 325 }}>
```
Comment (D7Torres) [inline documentation/content/components.progress-bar.md:27]: Just checking direction column and width 325 are intentional.

Comment (thomasdigby) [reply documentation/content/components.progress-bar.md:27]: Yep, this is a minor addition just to make it look a bit better in the preview

File: lib/src/components/top-bar/TopBar.mdx [hunk 4]
```diff
@@ -1,95 +0,0 @@
----
```
Comment (andoulla) [inline lib/src/components/top-bar/TopBar.mdx:1]: why is this deleted?

**Review by andoulla — APPROVED** [review]

**Review by D7Torres — APPROVED** [review]
Preemptive approval as far as Mariandi is satisfied with the answer you give her

File: lib/src/components/top-bar/TopBar.mdx (hunk 4 above)
Comment (thomasdigby) [reply lib/src/components/top-bar/TopBar.mdx:1]: It's unused, the _real_ TopBar documentation is stored in `/documentation/content/components.top-bar.md`. Just cleaning up
